spec/services/user_notification_mailer_service_spec.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe UserNotificationMailerService do
+  before { ENV['APPLICATION_EMAIL'] = 'contact@mailrandom.fr' }
+
+  describe 'send_yesterday_modification_notifications' do
+    subject(:send_user_notifications) { described_class.send_yesterday_modification_notifications }
+
+    before do
+      allow(UserMailer).to receive(:delay) { UserMailer }
+      allow(UserMailer).to receive(:yesterday_modifications)
+    end
+
+    context 'one selected assistance modified during the last 24h' do
+      let(:selected_assistance_expert1) { create :selected_assistance_expert }
+      let(:selected_assistance_expert2) { create :selected_assistance_expert }
+      let(:expected_array) do
+        [
+          {
+            expert_name: selected_assistance_expert2.expert_full_name,
+            expert_institution: selected_assistance_expert2.expert_institution_name,
+            question_title: selected_assistance_expert2.diagnosed_need.question_label,
+            company_name: selected_assistance_expert2.diagnosed_need.diagnosis.visit.facility.company.name_short,
+            start_date: selected_assistance_expert2.created_at.to_date,
+            old_status: 'quo',
+            current_status: 'done'
+          }
+        ]
+      end
+
+      before do
+        selected_assistance_expert1.update status: 'done', updated_at: Date.today - 2.days
+        Audited::Audit.last.update created_at: Date.today - 2.days
+
+        selected_assistance_expert2.update status: 'done'
```

Comment:
I would have prefered the symbol version (`:done`) but whatever.

Reply:
You could factorize this line and the 3 above in the first `before` of the file. Same with some `let` in each `context`.